lib/ansible/module_utils/k8s/raw.py

```diff
@@ -261,3 +296,56 @@ def create_project_request(self, definition):
         result['changed'] = True
         result['method'] = 'create'
         return result
+
+    def _wait_for(self, resource, name, namespace, predicate, timeout, state):
+        start = datetime.now()
+
+        def _wait_for_elapsed():
+            return (datetime.now() - start).seconds
+
+        response = None
+        while _wait_for_elapsed() < timeout:
+            try:
+                response = resource.get(name=name, namespace=namespace)
+                if predicate(response):
+                    return True, response.to_dict(), _wait_for_elapsed()
+                time.sleep(timeout // 20)
+            except NotFoundError:
+                if state == 'absent':
+                    return True, response.to_dict(), _wait_for_elapsed()
+        if response:
+            response = response.to_dict()
+        return False, response, _wait_for_elapsed()
+
+    def wait(self, resource, definition, timeout, state='present'):
+
+        def _deployment_ready(deployment):
+            # FIXME: frustratingly bool(deployment.status) is True even if status is empty
```

Comment:
We could update the bool evaluation on the result object in the client to fix this pretty easily

Reply:
That would be pretty useful

Reply:
However, relying on that fix would mean not supporting older versions - there's no reason why `wait` can't work with 0.6.0 for example. So while I'd welcome the fix, I won't be changing this.

Reply:
Good point